fe/fe-core/src/main/java/org/apache/doris/job/extensions/mtmv/MTMVTask.java

```diff
@@ -272,6 +285,220 @@ public void run() throws JobException {
+            }
+            // current, if partitionType is SELF_MANAGE, we can only FULL refresh
+            if (mtmv.getMvPartitionInfo().getPartitionType() == MTMVPartitionType.SELF_MANAGE) {
+                return Lists.newArrayList(mtmv.getPartitionNames());
+            }
+            // We need to use a newly generated relationship and cannot retrieve it using mtmv.getRelation()
+            // to avoid rebuilding the baseTable and causing a change in the tableId
+            return MTMVPartitionUtil.getMTMVNeedRefreshPartitions(context, relation.getBaseTablesOneLevel());
+        }
+
+        public List<String> calculateNeedRefreshPartitionsByManual(MTMVRefreshContext context) {
+            // check whether the user manually triggers it
+            if (taskContext.getTriggerMode() == MTMVTaskTriggerMode.MANUAL) {
+                if (taskContext.isComplete()) {
+                    return Lists.newArrayList(mtmv.getPartitionNames());
+                } else if (!CollectionUtils.isEmpty(taskContext.getPartitions())) {
+                    return taskContext.getPartitions();
+                }
+            }
+            return Lists.newArrayList();
+        }
+    }
+
+    private class MTMVFullDataRefreshExec extends MTMVDataRefreshExec {
+
+        @Override
+        protected void prepare(MTMVRefreshContext context,
+                Map<String, MTMVRefreshPartitionSnapshot> execPartitionSnapshots) {}
+
+        @Override
+        protected Command getDataRefreshCommand(
+                Set<String> refreshPartitionNames, Map<TableIf, String> tableWithPartKey) throws UserException {
+            return UpdateMvUtils
+                    .from(mtmv, mtmv.getMvPartitionInfo().getPartitionType() != MTMVPartitionType.SELF_MANAGE
+                            ? refreshPartitionNames : Sets.newHashSet(), tableWithPartKey, false, new HashMap<>());
+        }
+
+    }
+
+    private class MTMVIncrementalDataRefreshExec extends MTMVDataRefreshExec {
+        private boolean incremental;
+        private long tableSnapshotId;
+        private long mvSnapshotId;
+        private Map<String, String> params;
+
+        public void prepare(MTMVRefreshContext context,
+                Map<String, MTMVRefreshPartitionSnapshot> execPartitionSnapshots) throws Exception {
+            params = new HashMap<>();
+            Set<BaseTableInfo> baseTables = relation.getBaseTablesOneLevel();
+            if (baseTables.size() != 1) {
+                throw new JobException("Only support incremental refresh for single table MTMV");
+            }
+            BaseTableInfo baseTableInfo = baseTables.iterator().next();
+            MTMVSnapshotIf mvSnapshot = mtmv.getRefreshSnapshot().getMVSnapshot(mtmv.getName(), baseTableInfo);
+            mvSnapshotId = Optional.ofNullable(mvSnapshot).map(MTMVSnapshotIf::getSnapshotVersion).orElse(0L);
+            params.put(TableScanParams.READ_MODE, TableScanParams.INCREMENTAL_READ);
```

Comment:
I think we should extract a interface for this "params generation" logic. Because different datasource(like Paimon or Hudi) may have different parameter names.
So:
1. You move the the field like `DORIS_START_SNAPSHOT_ID` from `PaimonScanNode` to `TableScanParams`, I think you move them back because they are only for Paimon.
2. Use interface to separate the certain implementation of different data source.